services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -212,9 +216,62 @@ def engine(self) -> Engine:
             async for row in conn.execute(query):
                 result_dict = dict(row.items())
                 log.debug("found project: %s", result_dict)
+                tags = await self._get_tags_by_study(study_id=result_dict['id'])
+                result_dict['tags'] = tags
                 projects_list.append(_convert_to_schema_names(result_dict))
         return projects_list
 
+    async def _get_study(self, user_id: str, project_uuid: str, exclude_foreign: List = []) -> Dict: # pylint: disable=dangerous-default-value
+        async with self.engine.acquire() as conn:
+            joint_table = user_to_projects.join(projects)
+            query = select([projects]).select_from(joint_table).where(
+                and_(projects.c.uuid == project_uuid,
+                     user_to_projects.c.user_id == user_id)
+            )
+            result = await conn.execute(query)
+            study_row = await result.first()
+
+            if not study_row:
+                raise ProjectNotFoundError(project_uuid)
+
+            study = dict(study_row.items())
+
+            if 'tags' not in exclude_foreign:
```

Comment:
Can you please define exclude_foreign? I do not really understand it. You mean foreign keys?  You seem only to apply to tags... oder? why not create a flag called ``include_tags`` or something like this?

Reply:
so the idea is to query for related items by default. but in some cases one might prefer to avoid these extra queries if the information needed is already contained in the projects table. I prefer a list of entities that shouldn't be resolved over one flag per entity...